plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/controls/lightgrid/GridColumn.java

```diff
@@ -214,18 +214,31 @@ int computeHeaderWidth() {
         }
         {
             int textWidth;
-            if (Boolean.TRUE.equals(labelProvider.getGridOption(IGridLabelProvider.OPTION_EXCLUDE_COLUMN_NAME_FOR_WIDTH_CALC))) {
-                textWidth = grid.sizingGC.stringExtent("X").x;
-            } else {
-                String text = labelProvider.getText(this);
-                String description = labelProvider.getDescription(this);
-                textWidth = grid.sizingGC.stringExtent(text).x;
-                if (!CommonUtils.isEmpty(description)) {
-                    int descWidth = grid.sizingGC.stringExtent(description).x;
-                    if (descWidth > textWidth) {
-                        textWidth = descWidth;
+            Object calcWidthMethod = labelProvider.getGridOption(IGridLabelProvider.OPTION_CALC_COLUMN_WIDTH_METHOD);
+            String text = "X";
+            switch ((int) calcWidthMethod) {
+                case (0):
+                    text = labelProvider.getText(this);
+                    textWidth = grid.sizingGC.stringExtent(text).x;
+                    break;
+                case (1):
+                    textWidth = grid.sizingGC.stringExtent(text).x;
+                    break;
+                case (2):
```

Comment:
Can we create a enum for these values?

Reply:
Yes,  we can )
Btw, are you (your team) at pgConf now?

Reply:
No, we don't have a speaker at this conference this year :)

Reply:
But I can't say that none is there, as I don't know for sure